lib/active_record/connection_adapters/oracle_enhanced/jdbc_connection.rb

```diff
@@ -506,6 +506,8 @@ def get_ruby_value_from_result_set(rset, i, type_name, get_lob_value = true)
           else
             BigDecimal.new(d.stringValue)
           end
+        when :BINARY_FLOAT
+          rset.getFloat(i)
```

Comment:
This introduces a new bug: `NULL` values will be always casted to `0.0` since [getFloat](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getFloat(int)) never returns `nil`.

A quick fix could look like:

    rset.getString(i) && rset.getFloat(i)

(since [getString](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getString(int)) returns `nil` for `NULL` values)

Reply:
Created issue #2040 regarding this problem